gocd/bci_repo_publish.py

```diff
@@ -166,9 +186,13 @@ def run(self, version, token=None):
             }
             url = makeurl(self.apiurl, ['trigger', 'release'], params)
             # No bindings for using tokens yet, so do the request manually
-            req = requests.post(url, headers={'Authorization': f'Token {token}'})
-            if req.status_code != 200:
-                raise RuntimeError(f'Releasing failed: {req.text}')
+            req = requests.post(url, headers={'Authorization': f'Token {repo_token}'})
+            req.raise_for_status()
+
+        for pkg in self.fetch_package_names_in_project(build_prj):
+            url = makeurl(self.apiurl, ['trigger', 'rebuild'], {'project': build_prj, 'package': pkg})
```

Comment:
Sadly, a list doesn't work.
```ShellSession
❯ curl -H "Authorization: Token $TOKEN" -X POST "https://api.opensuse.org/trigger/rebuild?project=devel:BCI:SLE-15-SP5&package=rmt-helm,sle15-kernel-module-devel,rust-oldstable-image,rust-stable-image"
<status code="unknown_package">
  <summary>Package not found: devel:BCI:SLE-15-SP5/rmt-helm,sle15-kernel-module-devel,rust-oldstable-image,rust-stable-image</summary>
</status>
```

And specifying the parameter multiple times rebuilds just one package:
```ShellSession
❯ curl -H "Authorization: Token $TOKEN" -X POST "https://api.opensuse.org/trigger/rebuild?project=devel:BCI:SLE-15-SP5&package=rmt-helm&package=sle15-kernel-module-devel&package=rust-oldstable-image&package=rust-stable-image"
<status code="ok">
  <summary>Ok</summary>
</status>
```

So sadly, we need to DoS OBS :shrug:

Reply:
Dirk is most likely confusing it with the route used by rebuildpac. The rebuild token is a very special route

Reply:
Indeed. I didn't realize that this uses /trigger instead of /build . The letter has a project wide rebuild option.

I wonder if we somehow can give the bot access to that route.

Reply:
https://github.com/openSUSE/open-build-service/issues/2094 is the history of this route

Reply:
and https://github.com/openSUSE/open-build-service/pull/15119 is a new attempt at getting a fix in.